code/src/main/kotlin/com/expediagroup/sdk/core/http/Headers.kt

```diff
@@ -0,0 +1,207 @@
+            headersMap.computeIfAbsent(processHeaderName(name)) { mutableListOf() }.addAll(processHeaderValues(values))
+            return this
+        }
+
+        /**
+         * Sets the header with the specified name to the single value provided.
+         * If headers with this name already exist, they are removed.
+         *
+         * @param name the header name
+         * @param value the header value
+         * @return this builder
+         * @throws IllegalArgumentException if [name] or [value] is invalid
+         */
+        @Throws(IllegalArgumentException::class)
+        fun set(name: String, value: String): Builder {
+            return set(name, listOf(value))
+        }
+
+        /**
+         * Sets the header with the specified name to the values list provided.
+         * If headers with this name already exist, they are removed.
+         *
+         * @param name the header name
+         * @param values the header value
+         * @return this builder
+         * @throws IllegalArgumentException if [name] or [values] are invalid
+         */
+        @Throws(IllegalArgumentException::class)
+        fun set(name: String, values: List<String>): Builder {
+            val processedName = processHeaderName(name)
+
+            remove(processedName)
+            add(processedName, processHeaderValues(values))
+
+            return this
+        }
+
+        /**
+         * Removes any header with the specified name.
+         *
+         * @param name the header name
+         * @return this builder
+         */
+        fun remove(name: String): Builder {
+            headersMap.remove(name.lowercase(Locale.US))
+            return this
+        }
+
+        /**
+         * Builds an immutable [Headers] instance.
+         *
+         * @return the built [Headers]
+         */
+        fun build(): Headers {
+            return Headers(LinkedHashMap(headersMap))
+        }
+
+        private fun processHeaderName(name: String): String = name
+            .lowercase(Locale.US)
+            .trim()
+            .let {
+                validateHeaderName(it)
+                return@let it
+            }
+
+        private fun processHeaderValues(values: List<String>): List<String> = values
+            .map {
+                it.trim()
+            }
+            .let {
+                it.forEach { value -> validateHeaderValue(value) }
+                return@let it
+            }
+
+        private fun validateHeaderName(name: String) {
+            require(name.isNotBlank()) { "Header name must not be blank" }
+
+            for (char in name) {
+                require(char in '!'..'\u007E') {
+                    String.format("Invalid character %#04x in header name: %s", char.code, name)
+                }
+            }
+        }
+
+        private fun validateHeaderValue(value: String) {
+            for (char in value) {
+                require(char == '\t' || (char in '\u0020'..'\u007E')) {
```

Comment:
What are these ranges of invalid characters? Is it worth commenting, linking, or otherwise providing additional context?

Reply:
These are chars like `\n, \t, \r` ... I was thinking to remove these validators, it's very unlikely to happen, and if does, these will be caught by the underlying HTTP client. I'm with keeping these models as simple as possible

Reply:
If we can achieve the same functionality with zero code, it is better. DELETE!

Reply:
Done